manifests/init.pp

```diff
@@ -0,0 +1,61 @@
+# @summary
+#   Module to install Restic and manage backups
+#
+# @param user
+#   The user to create who owns the backup files.
+#
+# @param group
+#   The group to create who owns the backup files.
+#
+# @param package_manage
+#   Enable to manage the Restic package install.
+#
+# @param package_ensure
+#   The version of Restic to install.
+#
+# @param package_name
+#   The name of the Restic package to install.
+#
+class restic (
+  String $user = 'restic',
+  String $group = 'restic',
+  Boolean $package_manage = true,
+  String $package_ensure = latest,
+  String $package_name = 'restic',
+) {
+  $homedir = "/home/${user}"
+
+  if $package_manage {
+    package { $package_name:
+      ensure => $package_ensure,
+      before => File['/etc/restic'],
+    }
+  }
+
+  group { $group:
+    ensure => present,
+    system => true,
+  }
+
+  user { $user:
+    ensure => present,
+    gid    => $group,
+    system => true,
+    home   => $homedir,
+    shell  => '/usr/sbin/nologin',
```

Comment:
For now I only listed Debian support, but if we add RH support this should be probably `/sbin/nologin`. Though with the merged `/usr` it will work anyway.